code/obj/item/storage/secure_safe.dm

```diff
@@ -399,6 +399,46 @@ TYPEINFO(/obj/item/storage/secure/ssafe)
 					else // if what we selected wasn't a valid path
 						i++ // try again
 
+			if (10) // weaker power eggs
+				var/list/eggs_weak = list(/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/purple,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/mime,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/plant,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/candy,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/knight,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/wizard,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/ghost)
+
+				for (var/i=rand(1,rand(3,5)), i>0, i--)
+					var/egg = pick(eggs_weak)
+					if (ispath(egg))
+						src.storage.add_contents(new egg(src))
+				for (var/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/E in src.contents)
+					E.infertile = TRUE
+
+			if (11) // strong power eggs
+				var/list/eggs_strong = list(/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/robot,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/cockatrice,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/dream,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/popsicle)
+				#ifdef SECRETS_ENABLED
+					eggs_strong += list(/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/dragon,\
+					/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/coral,\
+					/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/phoenix,\
+					/obj/item/reagent_containers/food/snacks/ingredient/egg/chicken/zappy)
+				#endif
+				var/list/eggs_junk = list(/obj/item/reagent_containers/food/snacks/ingredient/egg/critter/nicespider,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/critter/spider,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/critter/clown,\
+				/obj/item/reagent_containers/food/snacks/ingredient/egg/critter/cluwne)
+				var/egg = pick(eggs_strong)
+				if (ispath(egg))
+					src.storage.add_contents(new egg(src))
+				for (var/i=rand(1,src.storage.slots), i>0, i--)
```

Comment:
this will overflow the storage if it rolls the max. maybe put in 2 strong, 2-4 junk?